java-checks/src/main/java/org/sonar/java/checks/tests/RandomizedTestDataCheck.java

```diff
@@ -0,0 +1,103 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks.tests;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.semantic.MethodMatchers;
+import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.NewClassTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+@Rule(key = "S5977")
+public class RandomizedTestDataCheck extends IssuableSubscriptionVisitor {
+
+  private static final MethodMatchers RANDOM_CONSTRUCTOR_METHOD_MATCHER = MethodMatchers.create()
+    .ofTypes("java.util.Random")
+    .constructor()
+    .withAnyParameters()
+      .build();
+
+  private static final MethodMatchers RANDOM_UUID_MATCHER = MethodMatchers.create()
+    .ofTypes("java.util.UUID")
+    .names("randomUUID")
+    .withAnyParameters()
+      .build();
+
+  private static final String LOCATIONS_TEXT = "usage of random data in test";
+  private static final String MESSAGE = "Replace randomly generated values with a fixed ones.";
+
+  private boolean reportedUUIDRandom = false;
+  private final List<Tree> randomSecondaryLocations = new ArrayList<>();
+
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return Arrays.asList(Tree.Kind.METHOD_INVOCATION, Tree.Kind.NEW_CLASS);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    if (tree.is(Tree.Kind.NEW_CLASS)) {
+      checkForRandomConstructorUsage(tree);
+    }
+    if (!reportedUUIDRandom && tree.is(Tree.Kind.METHOD_INVOCATION)) {
+      checkAndReportUUIDRandomUsage(tree);
+    }
+  }
+
+  private void checkForRandomConstructorUsage(Tree tree) {
+    NewClassTree newClassTree = (NewClassTree) tree;
+    if (RANDOM_CONSTRUCTOR_METHOD_MATCHER.matches(newClassTree)) {
+      randomSecondaryLocations.add(tree);
+    }
+  }
+
+  private void checkAndReportUUIDRandomUsage(Tree tree) {
+    MethodInvocationTree methodInvocationTree = (MethodInvocationTree) tree;
+    Symbol symbol = methodInvocationTree.symbol();
+    if (RANDOM_UUID_MATCHER.matches(methodInvocationTree)) {
+      reportedUUIDRandom = true;
+      List<JavaFileScannerContext.Location> locations = symbol.usages().stream()
+        .map(identifierTree -> new JavaFileScannerContext.Location(LOCATIONS_TEXT, identifierTree))
+        .collect(Collectors.toList());
+      reportIssue(tree, MESSAGE, locations, null);
+    }
+  }
+
+  @Override
+  public void leaveFile(JavaFileScannerContext context) {
```

Comment:
I tend to agree that reporting only one issue per test file is enough, we should just make sure that the [rspec](https://jira.sonarsource.com/browse/RSPEC-5977) description reflects this behavior (current Highlighting part mention that one issue per method will be reported).